### PR TITLE
Use correct version of dependabot-updater base image when running the 'updater' workflow

### PR DESCRIPTION
**Pull request by rhyskoedijk**
### What are you trying to accomplish?
The "updater" action/workflow should use the dependabot-updater base image that corresponds to the version of dependabot-omnibus set in `updater/Gemfile`. Currently "latest" will always be used, which can create problems as seen in https://github.com/tinglesoftware/dependabot-azure-devops/issues/1190#issuecomment-2212026759 

### Summary of changes
A new task has been added to `.github/workflows/updater.yml` that will read `updater/Gemfile`, extract the current version of dependabot-omnibus, find the git commit SHA for that version tag, then pass it to `updater/Dockerfile` as an argument.

In local/dev environment, the "latest" image tag is still used. Instructions have been added to the Dockerfile.

### Script testing
I have not tested this in a proper github action run, but have executed the script on a Ubunutu host.

#### When run on main branch
```log
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 15153  100 15153    0     0  30652      0 --:--:-- --:--:-- --:--:-- 30612
Dockerfile updated to use dependabot-updater image tag '24cc74a45208bb67bc8c1790ce2594f7940919bf' (v0.263.0)
```
Dockerfile:
```dockerfile
# The Dependabot docker images in https://github.com/dependabot/dependabot-core are no longer versioned like the Ruby Gems.
# In production, the build pipeline automatically updates the image tag to match the version in updater/Gemfile (see .github/workflows/updater.yml).
# In local/dev,  the "latest" tag will be used by default. You can override this by specifying a dependabot-core release tag commit SHA as the image tag.
#                e.g. for v0.264.0, use "ghcr.io/dependabot/dependabot-updater-$ECOSYSTEM:24cc74a45208bb67bc8c1790ce2594f7940919bf"
FROM ghcr.io/dependabot/dependabot-updater-$ECOSYSTEM:24cc74a45208bb67bc8c1790ce2594f7940919bf
```

#### When run on https://github.com/tinglesoftware/dependabot-azure-devops/pull/1191 branch
```log
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 15153  100 15153    0     0  91445      0 --:--:-- --:--:-- --:--:-- 91283
Dockerfile updated to use dependabot-updater image tag 'e8d8a1268ea61304e939ba9ab963e249cac5b241' (v0.264.0)
```
Dockerfile:
```dockerfile
# The Dependabot docker images in https://github.com/dependabot/dependabot-core are no longer versioned like the Ruby Gems.
# In production, the build pipeline automatically updates the image tag to match the version in updater/Gemfile (see .github/workflows/updater.yml).
# In local/dev,  the "latest" tag will be used by default. You can override this by specifying a dependabot-core release tag commit SHA as the image tag.
#                e.g. for v0.264.0, use "ghcr.io/dependabot/dependabot-updater-$ECOSYSTEM:24cc74a45208bb67bc8c1790ce2594f7940919bf"
FROM ghcr.io/dependabot/dependabot-updater-$ECOSYSTEM:e8d8a1268ea61304e939ba9ab963e249cac5b241
```

### CODE REVIEW

File: update-files.ps1
Comment (mburumaxwell) [inline update-files.ps1:1]: Hanged to this file seem unrelated to what we want to change. This particular section is useful for downloading the files from the core repo though we currently do not use them.

**Review by mburumaxwell — CHANGES_REQUESTED** [review]
This is great. However, instead of editing the Dockerfile, it may be better to use an argument with a default value then the workflow passes the argument like it does for `ECOSYSTEM`.

```docker
ARG BASE_VERSION=latest
# ...
FROM ghcr.io/dependabot/dependabot-updater-$ECOSYSTEM:$BASE_VERSION
```

You can output the computed/parsed version using

```bash
- name: "Get ..."
  id: docker-base-version
  run: |
    // ...
    echo "version={value}" >> $GITHUB_OUTPUT
```

Use the version:

```bash
--build-arg BASE_VERSION=${{ steps.docker-base-version.outputs.version }}
```

Let me know if this makes sense.

File: update-files.ps1
Comment (rhyskoedijk) [reply update-files.ps1:1]: that make sense, not sure why I didn't think of using an argument in the first place, I will update.

Do you want me to revert changes to `update-files.ps1`? I only edited it to remove any confusion or misunderstanding that the script is used to control the docker image version tags. Can undo this if you think it is still useful to keep this though

Comment (mburumaxwell) [reply update-files.ps1:1]: Let's leave the changes in the PowerShell script until we remove it all together in the future.

Comment (rhyskoedijk) [reply update-files.ps1:1]: I've switched to a Dockfile arg. Not sure if there is an easy want to test this, I wasn't able to run the actions/workflows from my fork so I haven't personally tested this all works.

Comment (mburumaxwell) [reply update-files.ps1:1]: The workflows run for the PR. Let me know if you can see them. We know it works because the correct base image is pulled for the current version (0.264.0)

```txt

#6 [1/7] FROM ghcr.io/dependabot/dependabot-updater-bundler:e8d8a1268ea61304e939ba9ab963e249cac5b241@sha256:9fe95dd189f43a8a3f8d5c287e77929546dbe84956be234ae20718ffbb6350a2

```

I guess this is enough proof?

**Review by mburumaxwell — APPROVED** [review]
